lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -215,10 +215,14 @@ sub register {
         $| = 1;
         select STDOUT;
         $| = 1;
-        { exec $self->{script} }
-        die "Could not execute $self->{script}: $!";
+	if ($self->{run}) {
+	  $self->{run}->();
```

Comment:
Cool!

The short explanation is that ```$code->($c)``` and ```$c->$code``` is exactly the same as long as $code is a reference to a sub. See http://shadow.cat/blog/matt-s-trout/madness-with-methods/ for more details.